src/shared/components/Spinner/index.tsx

```diff
@@ -1,27 +1,128 @@
 import React, { FunctionComponent } from 'react'
-import classNames from 'classnames'
-import styles from './spinner.pcss'
-export type SpinnerSize = 'small' | 'large'
+import styled, { css, keyframes } from 'styled-components'
+import { COLORS } from '~/shared/utils/styled'
+export type SpinnerSize = 'small' | 'large' | 'medium'
 export type SpinnerColor = 'green' | 'white' | 'gray' | 'blue'
 type Props = {
     size?: SpinnerSize
     color?: SpinnerColor
     className?: string
-    containerClassname?: string
 }
 
 const Spinner: FunctionComponent<Props> = ({
     size = 'small',
     color = 'green',
     className,
-    containerClassname,
 }: Props) => (
-    <div className={classNames(styles.container, containerClassname)}>
-        <span
-            className={classNames(className, styles[size], styles.spinner, styles[color])}
-        />
-        <span className={styles.screenReaderText}>Loading...</span>
-    </div>
+    <SpinnerContainer className={className}>
+        <SpinnerElement $size={size} $color={color} />
+        <ScreenReaderText>Loading...</ScreenReaderText>
+    </SpinnerContainer>
 )
 
 export default Spinner
+
+const spin = keyframes`
+  0% {
+    transform: rotate(0deg);
+  }
+
+  100% {
+    transform: rotate(360deg);
+  }`
+
+const SpinnerContainer = styled.div`
+    display: inline-flex;
+    align-self: center;
+`
+
+const ScreenReaderText = styled.span`
+    clip: rect(1px, 1px, 1px, 1px);
+    clip-path: inset(50%);
+    height: 1px;
+    width: 1px;
+    margin: -1px;
+    overflow: hidden;
+    padding: 0;
+    position: absolute;
+`
+
+const SpinnerElement = styled.span<{ $color: SpinnerColor; $size: SpinnerSize }>`
+    ${({ $color, $size }) => {
```

Comment:
Or, or

```tsx
function getSizeStyles({ $size }: { $size?: SpinnerSize }) { /* … */ }

function getColorStyles({ $color }: { $color?: SpinnerColor }) { /* … */ }

const SpinnerElement = styled.span<{ $color: SpinnerColor; $size: SpinnerSize }>`
    display: inline-block;
    position: relative;
    transform: translateZ(0);
    animation: ${spin} 1.1s infinite linear;
    ${getSizeStyles}
    ${getColorStyles}
`
```

Your "inline" approach does work. I'm trying to find a more straight forward way, or one with less complexity.